app/bin/tools/merge_users.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:gcloud/db.dart';
+
+import 'package:pub_dartlang_org/account/models.dart';
+import 'package:pub_dartlang_org/frontend/models.dart';
+import 'package:pub_dartlang_org/frontend/service_utils.dart';
+import 'package:pub_dartlang_org/history/models.dart';
+
+final _argParser = ArgParser()
+  ..addOption('source-user-id',
+      help: 'The userId of the User that will be removed.')
+  ..addOption('target-user-id',
+      help:
+          'The userId of the User that will take over the entries of the removed User.')
+  ..addFlag('help', abbr: 'h', defaultsTo: false, help: 'Show help.');
+
+Future main(List<String> args) async {
+  final argv = _argParser.parse(args);
+  if (argv['help'] as bool == true) {
+    print(
+        'Usage: dart merge_users.dart --source-user-id [uuid] --target-user-id [uuid]');
+    print(
+        'Merges one user into another (moving the ownership over to a new one).');
+    print(_argParser.usage);
+    return;
+  }
+
+  final sourceUserId = argv['source-user-id'] as String;
+  final targetUserId = argv['target-user-id'] as String;
+  if (sourceUserId == null ||
+      sourceUserId.isEmpty ||
+      targetUserId == null ||
+      targetUserId.isEmpty) {
+    print('Both user-id parameter must be set.');
+    print(_argParser.usage);
+    exit(-1);
+  }
+  if (sourceUserId == targetUserId) {
+    print('Use different user-ids.');
+    exit(-1);
+  }
+
+  await withProdServices(() async {
+    final sourceUser = (await dbService
+            .lookup<User>([dbService.emptyKey.append(User, id: sourceUserId)]))
+        .single;
+    final targetUser = (await dbService
+            .lookup<User>([dbService.emptyKey.append(User, id: targetUserId)]))
+        .single;
+    if (targetUser == null) {
+      print('Target user does not exists.');
+      exit(-1);
+    }
+    if (targetUser.oauthUserId == null) {
+      print('Target user does not have oauth user id.');
+      print('Please say "yes" if you still want to continue, CTRL+C to abort:');
+      final answer = stdin.readLineSync().toLowerCase();
+      if (answer != 'y' && answer != 'yes') {
+        exit(1);
+      }
+      print('YES');
```

Comment:
haha, that's some copy/pasta :)

Maybe, just print warning if oauthUserId isn't set, and always ask for confirmation instead of `await Future.delayed(Duration(seconds: 5));`.

This is a fairly destructive act, we should ask for confirmation...